swarm/network/stream/visualized_snapshot_sync_sim_test.go

```diff
@@ -81,10 +81,19 @@ func watchSim(sim *simulation.Simulation) (context.Context, context.CancelFunc)
 	)
 
 	go func() {
-		for d := range disconnections {
-			log.Error("peer drop", "node", d.NodeID, "peer", d.PeerID)
-			panic("unexpected disconnect")
-			cancelSimRun()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case d := <-disconnections:
+				if d.Error != nil {
+					log.Error("peer drop event error", "node", d.NodeID, "peer", d.PeerID, "err", err)
+				} else {
+					log.Error("peer drop", "node", d.NodeID, "peer", d.PeerID)
+				}
+				panic("unexpected disconnect")
```

Comment:
I see the `panic` was there before. Could we get rid of it?
As disconnect is an error, but I don't think it's a reason to panic and abort all test exections.

Reply:
Also, there were more problems with watchSim function. We do not need to wait for healthy kademlia after snapshot upload, and I moved disconnection watchers to sim.Run just as it is used on other tests